src/types.rs

```diff
@@ -213,7 +213,7 @@ impl From<DaHeader> for ClientMsg {
 		ClientMsg {
 			header_hash: hash,
 			block_num: header.number,
-			dimensions: Dimensions { rows, cols },
+			dimensions: Dimensions::new(rows, cols).unwrap(), // The dimensions should be valid in the header already
```

Comment:
You could replace `From<DaHeader>` by `TryFrom<DaHeader` in order to handle this.